### ISSUE TITLE
Microsoft.WinAppDriver.Appium.WebDriver 1.0.1-Preview version is it ok to use ?

### ISSUE THREAD

**Issue by SwathiCM** [microsoft/WinAppDriver#1685]
Microsoft.WinAppDriver.Appium.WebDriver 1.0.1-Preview version

Selenium Web driver package v3.8.0

Selenium Support package v3.8.0

We are using the above mentioned versions for our Windows desktop app UI Automation along with WinAppDriver v1.2.1.
We are just skeptical about using older versions. Does anyone has any idea about vulnerability issues with this?
Is it ok to use this older versions?

**Comment by anunay1** [comment]
Microsoft.WinAppDriver.Appium.WebDriver 1.0.1-Preview version is just a preview version and there is no support there after. Better to use Appium.Webdriver.

**Comment by SwathiCM** [comment]
When we tried with Appium Webdriver 5.0.0 beta version, we get the following error.
![AppimWebdriver_5 0 0_Issue](https://user-images.githubusercontent.com/100187474/155314083-5ec8b7ae-937b-4c66-a3e8-54461f381c73.png)

Can you please suggest if any other stable tool available for Windows Desktop application automation ?

**Comment by anunay1** [comment]
use 
```
            var desktopCapabilities = new AppiumOptions();
            desktopCapabilities.App =  "Your app";

```

This is a stable tool, I have been using it for past 4 years in all my windows application automation.

**Comment by SwathiCM** [comment]
We are getting below error when we used suggested code .
![BadCapabilities](https://user-images.githubusercontent.com/100187474/155498102-dd9e8bc0-ef15-48c6-852f-129e3ae005b3.png)


Can you please tell us the stable version which you are using ?
Can you please let us know the WinAppDriver, Selenium web driver and Appium web driver version ?

**Comment by anunay1** [comment]
Appium.Webdriver:4.3.1
Selenium.Support: 3.141.0
Selenium.WebDriver:3.141.0

And It seems your application has a splash screen? Read Wiki.

**Comment by SwathiCM** [comment]
Thanks for your quick response.
We also tried using the same versions but it looks sometimes its working and sometimes we get the following error.
![Error_4-3-1Appium](https://user-images.githubusercontent.com/100187474/155506812-717ba20f-eb9f-4514-8502-6b6cd412c5ab.PNG)

We tried below code snippet.
![Appium4-3-1](https://user-images.githubusercontent.com/100187474/155506956-78a8f8ef-a629-44b1-9705-1f896f1b2dc6.PNG)

Yes and it looks like this application has splash screen. Is there any issue using it?

**Comment by anunay1** [comment]
You your application has a splash screen then the window handle will change, You need to create a root session and then switch to the opened window, the below code will do that for you 

```
  var allProcesses2 = Process.GetProcesses();
            foreach (Process process in allProcesses2)
            {
                if (process.ProcessName.Contains("Your application"))
                {
                    mainWindowTitle = process.MainWindowTitle;
                }
            }

            var desktopCapabilities2 = new AppiumOptions();
            desktopCapabilities2.AddAdditionalOption("app", "Root");
            string WindowsApplicationDriverUrl2 = "http://127.0.0.1:4725";
            desktopSession2 = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl2), desktopCapabilities2);
            Assert.IsNotNull(desktopSession2);

            var RSWindow1 = desktopSession2.FindElement(By.Name(mainWindowTitle));
            var RSTopLevelWindowHandle1 = RSWindow1.GetAttribute("NativeWindowHandle");
            RSTopLevelWindowHandle1 = (int.Parse(RSTopLevelWindowHandle1)).ToString("x");

            var appcapabilities1 = new AppiumOptions();
            appcapabilities1.AddAdditionalOption("appTopLevelWindow", RSTopLevelWindowHandle1);
            appcapabilities1.AddAdditionalOption("newCommandTimeout", 300);

            driver1 = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), appcapabilities1, TimeSpan.FromSeconds(50));

```

**Comment by SwathiCM** [comment]
Thanks for the code snippet. We used the below snippet.
![AppiumRootSession](https://user-images.githubusercontent.com/100187474/155666226-21f41384-72c3-484d-937b-8cd98fc5c296.png)

We used below versions :
Appium Web driver v4.3.1
Selenium Web driver v3.141.0
Selenium Support v3.141.0

Only changes were, we need to launch the application window before running the script.
Thanks for the quick response and help. Using the above snippets, things are stable.

**Comment by anunay1** [comment]
Why are you launch the application the application separately? In the try block have the code to launch the application, which you know will throw an exception and in the catch block have the logic to switch to the opened window.

**Comment by SwathiCM** [comment]
Yeah that's true. We can use the same approach. Thank you so much !

**Comment by kv-naveen** [comment]
@SwathiCM  @anunay1 have you started using WinAppDriver pre-release version or still on 1.2.1

**Comment by anunay1** [comment]
No.

**Comment by anunay1** [comment]
No.

**Comment by anunay1** [comment]
No.
